src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java

```diff
@@ -64,4 +84,121 @@ public boolean publish(String projectID, String projectName, UUID conversionID)
+        } catch (IOException e) {
+            logger.error("Error sending request.");
+            throw e;
+        }
+    }
+
+    /**
+     * This method will send an invitation email to the target user
+     * @param email
+     * @return the invitation ID
+     */
+    public String invite(String email) throws IOException {
+        // check if the env variables has been set
+        if (EXPO_USERNAME == null || EXPO_PASSWORD == null || EXPO_ORGANIZATION_ID == null) {
+            logger.error("Expo user credential hasn't been set! Please check the environment variables.");
+            return null;
+        }
+        // first login to the expo
+        try{
+            login(EXPO_USERNAME, EXPO_PASSWORD);
+        } catch (IOException e) {
+            logger.error("Error sending invitation.", e.fillInStackTrace());
+            return null;
+        }
+        // send request to expo graphql endpoint
+        String statement = ExpoGraphqlStatement.createUserInvitationForAccount(EXPO_ORGANIZATION_ID, email, "[VIEW]");
+        URL queryURL = new URL(EXPO_GRAPHQL_API);
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("Accept", "application/json");
+        headers.put("expo-session", sessionString);
+        try {
+            JsonObject response = sendPostRequest(queryURL, headers, statement);
+            return response.get("data").getAsJsonObject()
+                                    .get("userInvitation").getAsJsonObject()
+                                    .get("createUserInvitationForAccount").getAsJsonObject()
+                                    .get("id").getAsString();
+        } catch (IOException e) {
+            logger.error("Error sending invitation: " + e.fillInStackTrace());
+            return null;
+        }
+    }
+
+    public boolean cancelInvitation(String invitationID) throws IOException {
+        if (sessionString.equals("")) {
+            logger.error("No session string found, please login first.");
+            return false;
+        }
+        String statement = ExpoGraphqlStatement.createCancelInvitationStatement(invitationID);
+        URL queryURL = new URL(EXPO_GRAPHQL_API);
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("Accept", "application/json");
+        headers.put("expo-session", sessionString);
+        try {
+            sendPostRequest(queryURL, headers, statement);
+            return true;
+        } catch (IOException e) {
+            logger.error("Error canceling invitation: " + e.fillInStackTrace());
+        }
+        return false;
+    }
+
+    /**
+     * This method will login to the expo.io and get the session object string
+     * @param username
+     * @param password
+     * @return
+     */
+    public JsonObject login(String username, String password) throws IOException {
```

Comment:
within `login` function I'd check if `sessionString` already exists (and potentially if it has valid `expiresAt` timestamp, but that is in my opinion redundant, as their sessionStrings expire at 10 years time), and if it does do not perform API call

Reply:
Good point, I think I can send a request to clear all other sessions after successfully logged in.